File: src/main/java/org/goafabric/dbagentnew/dbagent/persistence/entity/Address.java
package org.goafabric.dbagentnew.dbagent.persistence.entity;

import jakarta.persistence.*;


@Entity
@Table(name="address")
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Version //optimistic locking
    private Long version;

    private String street;
    private String city;

    public Address(String id, Long version, String street, String city) {
        this.id = id;
        this.street = street;
        this.city = city;
        this.version = version;
    }

    Address() {
    }

    public String getId() {
        return id;
    }

    public String getStreet() {
        return street;
    }

    public String getCity() {
        return city;
    }

    public Long getVersion() {
        return version;
    }

    @Override
    public String toString() {
        return "AddressEo{" +
                "id='" + id + '\'' +
                ", version=" + version +
                ", street='" + street + '\'' +
                ", city='" + city + '\'' +
                '}';
    }
}

File: src/main/kotlin/org/goafabric/llm/tool/persistence/entity/Address.kt
package org.goafabric.llm.tool.persistence.entity

import jakarta.persistence.*

@Entity
@Table(name = "address")
class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: String? = null
        private set

    @Version //optimistic locking
    var version: Long? = null
        private set

    var street: String? = null
        private set
    var city: String? = null
        private set

    constructor(id: String?, version: Long?, street: String?, city: String?) {
        this.id = id
        this.street = street
        this.city = city
        this.version = version
    }

    internal constructor()

    override fun toString(): String {
        return "AddressEo{" +
                "id='" + id + '\'' +
                ", version=" + version +
                ", street='" + street + '\'' +
                ", city='" + city + '\'' +
                '}'
    }
}
